### PR TITLE
feat(eslint): prepare migration to typescript-eslint 8.*

### PR DESCRIPTION
**Pull request by kpanot**
## Proposed change

prepare migration to typescript-eslint 8.*

## Related issues

- :rocket: Pull Request #2027

### CODE REVIEW

**Review by matthieu-crouzet — APPROVED** [review]

File: packages/@o3r/eslint-plugin/src/rules/utils.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import { ESLintUtils, TSESLint, TSESTree } from '@typescript-eslint/utils';
 import * as path from 'node:path';
+import { readFileSync } from 'node:fs';
+import { PackageJson } from 'type-fest';
```
Comment (vscaiceanu-1a) [inline packages/@o3r/eslint-plugin/src/rules/utils.ts:4]:
```suggestion
import type { PackageJson } from 'type-fest';
```

File: packages/@o3r/eslint-plugin/src/rules/typescript/o3r-categories-tags/o3r-categories-tags.ts [hunk 2]
```diff
@@ -19,8 +19,7 @@ export default createRule<Readonly<[O3rCategoriesTagsRuleOption, ...any]>, Messa
     fixable: 'code',
     type: 'problem',
     docs: {
-      description: 'Ensures that @o3rCategories and @o3rCategory are used with correct value',
-      recommended: 'strict'
+      description: 'Ensures that @o3rCategories and @o3rCategory are used with correct value'
```
Comment (sdo-1A) [inline packages/@o3r/eslint-plugin/src/rules/typescript/o3r-categories-tags/o3r-categories-tags.ts:22]:
```suggestion
      description: 'Ensures that @o3rCategories and @o3rCategory are used with a correct value'
```

OR

```suggestion
      description: 'Ensures that @o3rCategories and @o3rCategory are used with correct values'
```

File: packages/@o3r/eslint-plugin/src/rules/typescript/o3r-widget-tags/o3r-widget-tags.ts [hunk 3]
```diff
@@ -47,8 +47,7 @@ export default createRule<[Readonly<O3rWidgetTagsRuleOption>, ...any], O3rWidget
     fixable: 'code',
     type: 'problem',
     docs: {
-      description: 'Ensures that @o3rWidget and @o3rWidgetParam are used with correct value',
-      recommended: 'strict'
+      description: 'Ensures that @o3rWidget and @o3rWidgetParam are used with correct value'
```
Comment (sdo-1A) [inline packages/@o3r/eslint-plugin/src/rules/typescript/o3r-widget-tags/o3r-widget-tags.ts:50]:
```suggestion
      description: 'Ensures that @o3rWidget and @o3rWidgetParam are used with a correct value'
```
OR
```suggestion
      description: 'Ensures that @o3rWidget and @o3rWidgetParam are used with correct values'
```

File: packages/@o3r/eslint-plugin/src/rules/utils.ts [hunk 4]
```diff
@@ -1,8 +1,10 @@
 import { ESLintUtils, TSESLint, TSESTree } from '@typescript-eslint/utils';
 import * as path from 'node:path';
+import { readFileSync } from 'node:fs';
```
Comment (sdo-1A) [inline packages/@o3r/eslint-plugin/src/rules/utils.ts:3]: should be above previous import? (alphabetical)

File: packages/@o3r/eslint-plugin/src/rules/yaml/yarnrc-package-extensions-harmonize/yarnrc-package-extensions-harmonize.ts [hunk 5]
```diff
@@ -32,8 +32,7 @@ export default createRule<[Options, ...any], 'versionUpdate' | 'error'>({
     hasSuggestions: true,
     type: 'problem',
     docs: {
-      description: 'Ensure that the package extension versions are aligned with range defined in packages.',
-      recommended: 'strict'
+      description: 'Ensure that the package extension versions are aligned with range defined in packages.'
```
Comment (sdo-1A) [inline packages/@o3r/eslint-plugin/src/rules/yaml/yarnrc-package-extensions-harmonize/yarnrc-package-extensions-harmonize.ts:35]:
```suggestion
      description: 'Ensure that the package extension versions are aligned with the range defined in the packages.'
```

File: packages/@o3r/eslint-plugin/src/rules/json/json-dependency-versions-harmonize/json-dependency-versions-harmonize.ts [hunk 6]
```diff
@@ -41,8 +41,7 @@ export default createRule<[Options, ...any], 'versionUpdate' | 'error'>({
     hasSuggestions: true,
     type: 'problem',
     docs: {
-      description: 'Ensure that the package dependency versions are aligned with the other package of the workspace.',
-      recommended: 'strict'
+      description: 'Ensure that the package dependency versions are aligned with the other package of the workspace.'
```
Comment (sdo-1A) [inline packages/@o3r/eslint-plugin/src/rules/json/json-dependency-versions-harmonize/json-dependency-versions-harmonize.ts:44]:
```suggestion
      description: 'Ensure that the package dependency versions are aligned with the other packages of the workspace.'
```
or leave package singular depending on the use case?

**Review by sdo-1A — APPROVED** [review]

**Pull request merged**